packages/cli/src/services/deployService.js

```diff
@@ -44,19 +46,45 @@ const upload = (url, data, apiKey, onUploadProgress) => {
         try {
           const parsedData = JSON.parse(rawData);
 
-          if (res.statusCode >= 200 && res.statusCode <= 300) {
+          if (res.statusCode >= 200 && res.statusCode < 300) {
             resolve(parsedData);
+          } else if (res.statusCode >= 300 && res.statusCode < 400) {
+            reject(`The server attempted an unsupported redirect. [${res.statusCode}]`);
+          } else if (res.statusCode >= 400 && res.statusCode < 500) {
+            switch (res.statusCode) {
+              case 400:
+                reject("Error: Failed to deploy. [400]");
+                break;
+              case 401:
+                reject(`Error: The API key was not accepted by ${url} [401]`);
+              default:
+                reject(`Error: Failed to deploy with an unknown error. [${res.statusCode}]`);
+            }
+          } else if (res.statusCode >= 500 && res.statusCode < 600) {
+            switch (res.statusCode) {
+              case 500:
+                reject(`Error: The SPAship has encountered a mysterious problem; someone call Richard Feynman! [500]`);
+                break;
+              case 501:
+                reject(
+                  `Error: The spaship CLI attempted an action not supported by the server, possible version mismatch. [501]`
+                );
+                break;
+              default:
+                reject(`Error: Unknown. [${res.statusCode}]`);
+            }
           } else {
-            reject(parsedData);
           }
         } catch (e) {
-          reject(e);
+          reject(
+            "Error: the server returned an invalid message.  The server may be down, or your .spashiprc.yml may be pointing at the wrong server."
```

Comment:
I think we should show the error details. (like when user using --verbose ?)

Reply:
Hm, for many of these I agree, but the one highlighted here is the error that occurs when the JSON is invalid.  Hence, the error message isn't going to be all that helpful.

What it says: `Uncaught SyntaxError: Unexpected token < in JSON at position 0`
What it means: `The server may be down, or your .spashiprc.yml may be pointing at the wrong server.`

My vote would be to leave this as-is.  I do agree that if we had a `--verbose` flag like you mentioned, this would be a good place to use it.  If someone adds `--verbose`, then we can expand this (and all the other) error messages to include original message & stack traces.

Reply:
Make senses